src/async_wrap.cc

```diff
@@ -222,10 +230,15 @@ void AsyncWrap::EmitTraceEventAfter(ProviderType type, double async_id) {
 
 
 void AsyncWrap::EmitAfter(Environment* env, double async_id) {
+  Isolate* isolate = env->isolate();
+  v8::Local<v8::Context> context = env->isolate()->GetCurrentContext();
+
   // If the user's callback failed then the after() hooks will be called at the
   // end of _fatalException().
   Emit(env, async_id, AsyncHooks::kAfter,
        env->async_hooks_after_function());
+
+  context->SetEmbedderData(CURRENT_RESOURCE_FIELD, v8::Null(isolate));
```

Comment:
If there was an uncaught exception that was handled by an `uncaughtException` handles, this isn’t reached. For the async ID stack, we currently handle that case by using `clear_async_id_stack()` in the C++ code/`clearAsyncIdStack()` on the JS side

Reply:
Can you make an example?

Reply:
@mcollina I think this only becomes observable after we clean up the push/pop thing, I’ll do it then